controller/share_manager_controller.go

```diff
@@ -618,8 +666,49 @@ func (c *ShareManagerController) syncShareManagerVolume(sm *longhorn.ShareManage
 	return nil
 }
 
+func (c *ShareManagerController) clearShareManagerLeaseHolder(sm *longhorn.ShareManager) error {
+	log := getLoggerForShareManager(c.logger, sm)
+
+	lease, err := c.ds.GetLease(sm.Name)
+	if err != nil && !apierrors.IsNotFound(err) {
+		log.WithError(err).Warn("Failed to retrieve lease for share manager from datastore")
+		return err
+	}
+
+	if lease != nil {
+		holder := *lease.Spec.HolderIdentity
+		if holder != "" {
+			log.Infof("Clearing lease holder %v for share manager.", holder)
+			*lease.Spec.HolderIdentity = ""
+			*lease.Spec.AcquireTime = metav1.MicroTime{Time: time.Now()}
```

Comment:
Could you explain why we need to reset the AcquireTime?

Reply:
It's not necessary logically, but I do set it whenever the holder identity changes.  In this case, it is handy for measuring how long the volume has taken to be brought back up after it is recognized as down.

Reply:
> In this case, it is handy for measuring how long the volume has taken to be brought back up after it is recognized as down.

I am not sure how this work because the share-manager pod also set it when it start here https://github.com/longhorn/longhorn-share-manager/pull/222/files#diff-35f2cbdb0cb5f8088b0c5eed774550415da6195bc3f0c7b504cb6e4bb51c145bR293-R294

Reply:
The share manager sets `AcquireTime` when it takes the lease, changing the `HolderIdentity`.  The rule still holds - `AcquireTime` is the time of last change to holder, whether by share-manager pod to set it, or share-manager controller to clear it.

Reply:
This thread is obsolete, since the `AcquireTime` has been commandeered to use as a flag for delinquency checks.